packagedb/api.py

```diff
@@ -749,18 +750,22 @@ def resolve_versions(parsed_purl, vers):
 
     all_versions = get_all_versions(parsed_purl) or []
 
-    return [
-        str(
-            PackageURL(
-                type=parsed_purl.type,
-                namespace=parsed_purl.namespace,
-                name=parsed_purl.name,
-                version=version.string,
-            )
-        )
-        for version in all_versions
-        if version in version_range
-    ]
+    result = []
+    for version in all_versions:
+        try:
+            if version in version_range:
+                package_url = PackageURL(
+                    type=parsed_purl.type,
+                    namespace=parsed_purl.namespace,
+                    name=parsed_purl.name,
+                    version=version.string,
+                )
+                result.append(str(package_url))
+        except Exception:
+            # Skip the ``Invalid constraints sequence`` Exception
```

Comment:
Can you explain this exception that is mentioned here? It seems heavy-handed to pass exceptions like this

Reply:
On the univers side, the method used to test whether a given version is present in the version range raises an exception when the version constraints in the version range is ambiguous. https://github.com/nexB/univers/blob/2f2c21aca9766978bbf1d8e17da9a513372eb1f9/src/univers/version_constraint.py#L539-L543

Reply:
Ack, I see why it is needed. I think it would be good to update the code at https://github.com/nexB/univers/blob/2f2c21aca9766978bbf1d8e17da9a513372eb1f9/src/univers/version_constraint.py#L539-L543 to return a specific exception subclass instead of Exception, so we can avoid capturing exceptions that are not for invalid constraints sequence.

Reply:
Added fix here https://github.com/nexB/univers/pull/124

Reply:
@JonoYang fixed, we're using `InvalidConstraintsError` instead of the generic Exception.